kube-client/src/client/auth/mod.rs

```diff
@@ -206,11 +206,10 @@ impl RefreshableToken {
             }
 
             RefreshableToken::File(token_file) => {
-                let mut locked = token_file.lock().await;
-                if let Some(token) = locked.cached_token() {
-                    bearer_header(token)
+                if let Some(header) = { let guard = token_file.read().await; guard.cached_token().map(bearer_header) } {
```

Comment:
hah, aggressive scoping. i think this makes sense, but will double check with controller-rs just in case.

it might make sense to put a comment on this line to explain the scope, or maybe see if we can put a unit test around `to_header` in the ::File case